### PR TITLE
Alex kooper/ipfs

### PR DESCRIPTION
**Pull request by alex-kooper**
This PR implements IPFS upload functionality. The following is implemented:
- Image upload in "Create a Non-Fungible Token" screen
- IPFS Cid(Hash) is read-only field, populated when you upload an image
- When run `bin/start` it automatically configures and starts IPFS server locally (using ipfs/go-ipfs docker image)

### CODE REVIEW

File: client/src/api/ipfsUploader.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const reader = new FileReader();
+    
+    reader.onload = () => { resolve(reader.result as ArrayBuffer); };
+    reader.onerror = () => { reject(reader.error); reader.abort(); };
+
+    reader.readAsArrayBuffer(blob);
+  })
+)
+
+const ipfsClient = IpfsClient('http://localhost:5001')
```
Comment (emishur) [inline client/src/api/ipfsUploader.ts:17]: Hardcoded URL instead of reading it from the config

File: client/src/api/ipfsUploader.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+const uploadToIpfs = async (blob: Blob): Promise<IpfsContent> => {
+  const buffer = await readBlobAsArrayBuffer(blob);
+  const ipfsFile = await ipfsClient.add(buffer);
+  
+  return {
+    cid: ipfsFile.cid.toString(),
+    size: ipfsFile.size,
+    url: `http://127.0.0.1:8080/ipfs/${ipfsFile.cid}`,
```
Comment (emishur) [inline client/src/api/ipfsUploader.ts:40]: Hardcoded host URL

File: bin/start [hunk 3]
```diff
@@ -34,4 +34,6 @@ $PROJECT_ROOT_DIR/bin/wait-for-postgres $PROJECT_ROOT_DIR/bin/migrate-db
 
 docker stack deploy -c $API_SERVER_STACK_FILE $STACK_NAME
 
+${BASH_SOURCE%/*}/start-ipfs
+
```
Comment (itkach) [inline bin/start:38]: Other script invocations in the same script use `$PROJECT_ROOT_DIR/bin/`, perhaps this one should as well - for consistency and bevcause PROJECT_ROOT_DIR is clearer than ${BASH_SOURCE%/*}

File: client/src/api/ipfsUploader.ts (hunk 1 above)
Comment (alex-kooper) [reply client/src/api/ipfsUploader.ts:17]: Config will come in the next PR. I am still figuring out how to use it properly.

File: client/src/api/ipfsUploader.ts (hunk 2 above)
Comment (alex-kooper) [reply client/src/api/ipfsUploader.ts:40]: Again, that is by design, still not sure how to use config, will come in the next PR.

File: bin/start (hunk 3 above)
Comment (alex-kooper) [reply bin/start:38]: Agree, will fix

**Pull request merged**